apps/examples/e2e/tests/test-ui.spec.ts

```diff
@@ -1,10 +1,109 @@
-import test from '@playwright/test'
+import test, { expect } from '@playwright/test'
 import { setup } from '../shared-e2e'
 
-test.describe('ui', () => {
+test.describe('mobile ui', () => {
 	test.beforeEach(setup)
-
-	test('mobile style panel opens and closes when tapped', async ({ isMobile }) => {
+	test('mobile style panel opens and closes when tapped', async ({ isMobile, page }) => {
+		test.skip(!isMobile, 'only run on mobile')
+		expect(await page.isVisible('.tlui-style-panel')).toBe(false)
+		await page.getByTestId('mobile-styles.button').click()
+		expect(await page.isVisible('.tlui-style-panel')).toBe(true)
+		await page.getByTestId('mobile-styles.button').click()
+		expect(await page.isVisible('.tlui-style-panel')).toBe(false)
+	})
+	test('mobile style panel closes when tapping off the menu', async ({ isMobile, page }) => {
 		test.skip(!isMobile, 'only run on mobile')
+		await page.getByTestId('mobile-styles.button').click()
+		expect(await page.isVisible('.tlui-style-panel')).toBe(true)
+		await page.getByTestId('tools.select').click()
```

Comment:
I'd also extract all the tools locators. Maybe we can use something like
```
const TOOLS = {
    select: 'tools.select',
    draw: `tools.draw,` 
} as const
```

We can then use them in tests like so
```
const selectTool = page.getByTestId(TOOLS.select)

```